config/main.py

```diff
@@ -5118,6 +5118,22 @@ def unbind(ctx, interface_name):
     for ipaddress in interface_ipaddresses:
         remove_router_interface_ip_address(config_db, interface_name, ipaddress)
     if table_name == "VLAN_SUB_INTERFACE":
+        # First delete subinterface, once subinterface deletion successful,
+        # recreate same with same config on default vrf
+        if 'state_db' not in ctx.obj:
+            if ctx.obj['namespace'] is DEFAULT_NAMESPACE:
+                state_db = SonicV2Connector(use_unix_socket_path=True)
+            else:
+                state_db = SonicV2Connector(use_unix_socket_path=True, namespace=ctx.obj['namespace'])
+            state_db.connect(state_db.STATE_DB, False)
+        else:
+            state_db = ctx.obj['state_db']
+
+        config_db.set_entry(table_name, interface_name, None)
+        _hash = '{}{}'.format('INTERFACE_TABLE|', interface_name)
+        while state_db.exists(state_db.STATE_DB, _hash):
```

Comment:
@prsunny It is not updated bur rather checked here to confirm that None has been processed and state_db is removed. We already have such loginc here.
https://github.com/sonic-net/sonic-utilities/blob/ce1c44496603cd895d02a33bb0fcacd33a5f41bf/config/main.py#L5105

Reply:
Thats right Prince. This is present since VRF bind() CLI was implemented as Sudharsan pointed.
Here click cli is ensuring STATE_DB INTERFACE_TABLE for the interface is deleted to ensure RIF for corresponding interface is deleted before binding interface to new VRF. In unbind scenario for subinterfaces, subinterface has to be deleted and recreated with same subinterface attributes in default vrf.

Reply:
my concern is what if there is an issue with updating state_db, this will be indefinitely hanging, correct?

Reply:
@prsunny , yes this will be hung indefinitely. We can handle this in intfmgrd. Can we track this as a separate issue since its applicable to all interfaces(vrf bind operation) and not specific to subinterfaces?